pnpm_10: 10.15.1 -> 10.17.0

gepbird opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Releases:
- https://github.com/pnpm/pnpm/releases/tag/v10.16.0
- https://github.com/pnpm/pnpm/releases/tag/v10.16.1
- https://github.com/pnpm/pnpm/releases/tag/v10.17.0

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

gepbird committed:
pnpm_10: 10.15.1 -> 10.17.0

Scrumplex commented:
https://github.com/pnpm/pnpm/issues/9966

Seems like we can fix this by setting `CI=true`. Not sure if we want to do this for every derivation, or just the cases where it's needed.

It could cause unexpected build changes I think.

Perhaps changing any occurrence of `pnpm prune` to `CI=true pnpm prune` would be good enough

tjenkinson commented:
👋 hey I added this error to pnpm. Do you have some logs where it was working before? I'm not convinced the prune would actually have been working

gepbird commented:
The overridden `vips` required by `zipline` is also failing on master, `n8n` just takes too long.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 442505 --skip-package n8n`
Commit: `28132d0470643439bfca12e151295d21df86babc`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>zipline</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 55 packages built:</summary>
  <ul>
    <li>apache-answer</li>
    <li>authelia</li>
    <li>bash-language-server</li>
    <li>bumpp</li>
    <li>equibop</li>
    <li>equicord</li>
    <li>fedistar</li>
    <li>folo</li>
    <li>froide (python313Packages.froide)</li>
    <li>froide-govplan</li>
    <li>froide-govplan.dist</li>
    <li>froide.dist (python313Packages.froide.dist)</li>
    <li>gale</li>
    <li>gitify</li>
    <li>heroic</li>
    <li>heroic-unwrapped</li>
    <li>home-assistant-custom-lovelace-modules.custom-sidebar</li>
    <li>homepage-dashboard</li>
    <li>homer</li>
    <li>immich</li>
    <li>immich-cli</li>
    <li>legcord</li>
    <li>memos</li>
    <li>moonlight</li>
    <li>ni</li>
    <li>openlist</li>
    <li>paperless-ngx</li>
    <li>paperless-ngx.dist</li>
    <li>pnpm (pnpm_10)</li>
    <li>pocket-id</li>
    <li>podman-desktop</li>
    <li>python312Packages.django-filingcabinet</li>
    <li>python312Packages.django-filingcabinet.dist</li>
    <li>python312Packages.froide</li>
    <li>python312Packages.froide.dist</li>
    <li>python313Packages.django-filingcabinet</li>
    <li>python313Packages.django-filingcabinet.dist</li>
    <li>renovate</li>
    <li>sftool-gui</li>
    <li>signal-desktop</li>
    <li>splayer</li>
    <li>teleport (teleport_17)</li>
    <li>teleport.client (teleport_17.client)</li>
    <li>teleport_16</li>
    <li>teleport_16.client</li>
    <li>teleport_18</li>
    <li>teleport_18.client</li>
    <li>typespec</li>
    <li>umami</li>
    <li>vencord</li>
    <li>vencord-web-extension</li>
    <li>vesktop</li>
    <li>vue-language-server</li>
    <li>webcord-vencord</li>
    <li>youtube-music</li>
  </ul>
</details>

gepbird commented:
> 👋 hey I added this error to pnpm. Do you have some logs where it was working before? I'm not convinced the prune would actually have been working

Thanks for coming, one of the logs are available at https://cache.nixos.org/log/73xcb80m443dnwa07s92mirbizi0y1kg-vue-language-server-3.0.6.drv. It was built with pnpm 10.15.1, I think the output of `pnpm prune --prod` starts at "The modules directories will be removed and reinstalled from scratch", I'll check if it actually pruned it, because it showed a "Procceed (y/n)" prompt.

gepbird commented:
I added `CI=true` to `pnpm.configHook` and will rebuild all the packages in about a day.

tjenkinson commented:
Hey @gepbird yeh the `true` after is because that's the default option. It wasn't actually chosen though. The bug would cause pnpm to just exit

gepbird commented:
@tjenkinson Thanks for the help!

Slightly off topic, but do you know anything about https://github.com/pnpm/pnpm/issues/5315 ? I think fixing it would greatly help us, because AFAIK it would put all the node_modules the application needs into a single directory.

In other words, we could replace this:
```sh
    pnpm prune --prod
    find -type f \( -name "*.ts" -o -name "*.map" \) -exec rm -rf {} +

    # https://github.com/pnpm/pnpm/issues/3645
    find node_modules packages/language-server/node_modules -xtype l -delete

    # remove non-deterministic files
    rm node_modules/.modules.yaml node_modules/.pnpm-workspace-state-v1.json
```

with just this:
```sh
    pnpm deploy --offline --prod # and maybe --filter=...
```
```

tjenkinson commented:
Hmm no sorry we don't use `pnpm deploy` atm so haven't hit that. May take a look though

gepbird commented:
> @gepbird have you tried `deploy` recently? [pnpm/pnpm#5315 (comment)](https://github.com/pnpm/pnpm/issues/5315#issuecomment-3288353034)

I just tried it with `vue-language-server`, it doesn't have the `injectWorkspacePackages: true` setting, so I ran with `--legacy`and got "ERR_PNPM_NO_OFFLINE_META  Failed to resolve typescript@* in package mirror /build/tmp.UzeNGefPhA/.cache/pnpm/metadata-v1.3/registry.npmjs.org/typescript.json". When I'll have more time, I'll try again with `injectWorkspacePackages: true`.

friedow reviewed (approved):
Thanks for fixing this <3

gepbird committed:
treewide: set CI=true for pnpm prune

In a recent PNPM version `pnpm prune` started creating a confirmation
prompt which was automatically denied, as there was no TTY.
Adding CI environment variable will make sure there are no prompts.

AdrienLemaire commented:
@gepbird I tried twice to `nixpkgs-review pr 442505`, waiting 45'~1h and killing all other processes (firefox, discord etc), but my system ended freezing both times, leaving me no other option than force-shutting down the laptop. Running on Dell XPS 13 9370.
Giving up on this review, sorry.

eclairevoyant commented:
This is a breaking change and should not be backported.

gepbird commented:
I tested all the packages where I added `CI=true`, only `vue-language-server` is broken at first sight. Other packages behave like on master (either work by opening a GUI or a web server, language servers ask for extra arguments or programs requiring a config file).
Since 25.11 is not far away, I'm not too attached to backport this with a fix for vue-language-server, but there shouldn't be other breakages.
